luarules/gadgets/unit_infestor_spawn.lua

```diff
@@ -19,23 +19,13 @@ end
 
 
 
-local isBuilding = {}
-local isInfestor = {}
-for udid, ud in pairs(UnitDefs) do
-	if string.find(ud.name, 'leginfestor') then
-		isInfestor[udid] = true
-		Spring.Echo(udid)
-	end
-	if ud.isBuilding then
-		isBuilding[udid] = true
-	end
-end
-
+local infestorId = UnitDefNames.leginfestor.id
 
 function gadget:UnitCreated(unitID, unitDefID, unitTeam, builderID)
-	if builderID and isInfestor[unitDefID] and isInfestor[Spring.GetUnitDefID(builderID)] then
-		local OrderUnit = Spring.GiveOrderToUnit
-		OrderUnit(unitID, CMD.GUARD, { builderID },            { "shift" })
-	end
+    if (builderID) then
+        if (infestorId == unitDefID) and (Spring.GetUnitDefID(builderID) == infestorId) then
+            local OrderUnit = Spring.GiveOrderToUnit
+            OrderUnit(unitID, CMD.GUARD, { builderID }, { "shift" })
```

Comment:
since `UnitCreated` will be getting called very frequently, it would be good to also cache these engine functions at the top of the file
```
spGetUnitDefId = Spring.GetUnitDefID
spGiveOrderToUnit = Spring.GiveOrderToUnit
```

Reply:
Done now, and on liche change also